packages/blockchain-wallet-v4/src/redux/walletSync/middleware.js

```diff
@@ -37,15 +38,15 @@ export const getHDAccountAddressPromises = curry((state, account) => {
   const asyncDerive = index => toAsync(() =>
     HDAccount.getReceiveAddress(account, index, networks.bitcoin.NETWORK_BITCOIN))
   return selectors.data.bitcoin.getReceiveIndex(xpub, state)
-    .map((receiveIndex) => range(receiveIndex, receiveIndex + unusedAddressesCount))
+    .map((receiveIndex) => range(receiveIndex, receiveIndex + addressLookaheadCount))
```

Comment:
This still won't include your labeled addresses. You need to select the labeled address indices from the state and derive those as well as what you're already doing.